helpdesk_mgmt_rating/models/helpdesk_ticket.py

```diff
@@ -0,0 +1,83 @@
+from odoo import _, api, fields, models
+from odoo.tools.safe_eval import safe_eval
+
+
+class HelpdeskTicket(models.Model):
+    _name = "helpdesk.ticket"
+    _inherit = ["helpdesk.ticket", "rating.mixin"]
+
+    positive_rate_percentage = fields.Integer(
+        string="Positive Rates Percentage",
+        compute="_compute_percentage",
+        store=True,
+        default=-1,
+    )
+
+    rating_status = fields.Selection(
+        selection=[
+            ("stage_change", "Rating when changing stage"),
+            ("no_rate", "No rating"),
+        ],
+        string="Customer Rating",
+        default="stage_change",
+        required=True,
+    )
+
+    @api.depends("rating_ids.rating")
+    def _compute_percentage(self):
+        for ticket in self:
+            activity = ticket.rating_get_grades()
+            ticket.positive_rate_percentage = (
+                activity["great"] * 100 / sum(activity.values())
+                if sum(activity.values())
+                else -1
+            )
+
+    def write(self, vals):
+        res = super().write(vals)
+        if "stage_id" in vals and vals.get("stage_id"):
+            stage = self.env["helpdesk.ticket.stage"].browse(vals.get("stage_id"))
+            if stage.rating_mail_template_id:
+                self._send_ticket_rating_mail(force_send=False)
+        return res
+
+    def _send_ticket_rating_mail(self, force_send=False):
+        for ticket in self:
+            if ticket.rating_status == "stage_change":
+                survey_template = ticket.stage_id.rating_mail_template_id
+                if survey_template:
+                    ticket.rating_send_request(
+                        survey_template,
+                        lang=ticket.partner_id.lang,
+                        force_send=force_send,
+                    )
+
+    def rating_apply(self, rate, token=None, feedback=None, subtype=None):
+        return super().rating_apply(
+            rate,
+            token=token,
+            feedback=feedback,
+            subtype="helpdesk_mgmt_rating.mt_ticket_rating",
+        )
```

Comment:
When giving a rating it gives an error, in v14 the rating parameter has another name.
https://github.com/odoo/odoo/blob/7015dcd56566e93418f51e8e3562f384d10b4ffa/addons/rating/models/rating_mixin.py#L172

```suggestion
    def rating_apply(self, rate, token=None, feedback=None, subtype_xmlid=None):
        return super().rating_apply(
            rate,
            token=token,
            feedback=feedback,
            subtype_xmlid="helpdesk_mgmt_rating.mt_ticket_rating",
        )
```

Reply:
Thanks for the comment, I've fixed it and several other things